src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -485,11 +485,22 @@ bool Unit::IsWithinMeleeRange(Unit const* obj) const
     float dz = GetPositionZMinusOffset() - obj->GetPositionZMinusOffset();
     float distsq = dx*dx + dy*dy + dz*dz;
 
-    float maxdist = GetCombatReach() + obj->GetCombatReach() + 4.0f / 3.0f;
+    float maxdist = GetMeleeRange(obj);
 
     return distsq <= maxdist * maxdist;
 }
 
+float Unit::GetMeleeRange(Unit const* target) const
+{
+    float range = this->GetCombatReach() + 4.0f / 3.0f;
+    if (target)
+        range += target->GetCombatReach();
+    else
```

Comment:
The cases where `Unit::GetMeleeRange(NULL)` is called does not come from `Unit::IsWithinMeleeRange `but from `Spell::GetMinMaxRange `which also make use of the function. But yeah, that part is a bit strange to explain, so I suppose it should be refactored.